### PR TITLE
Refactor ConcurrentExecPool

### PR DESCRIPTION
**Pull request by bernhardmgruber**
TODO:

- [x] Merge #1998 first

### CODE REVIEW

**Comment by fwyzard** [comment]
ooohhhh... 2000 !

**Comment by psychocoderHPC** [comment]
@bernhardmgruber This PR can be rebased

**Review by psychocoderHPC — APPROVED** [review]

File: include/alpaka/core/ThreadPool.hpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+#pragma once
+
+#include "alpaka/core/Common.hpp"
+
+#include <atomic>
+#include <future>
+#include <mutex>
+#include <queue>
+#include <vector>
```
Comment (psychocoderHPC) [inline include/alpaka/core/ThreadPool.hpp:14]: `#inlcude <optional>` is missing

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: include/alpaka/core/ThreadPool.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply include/alpaka/core/ThreadPool.hpp:14]: Thx!

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**